### ISSUE TITLE
DPL-829 PBMC plate --> tube rack bed verification

### ISSUE THREAD

**Issue by KatyTaylor** [sanger/limber#1333]
**User story**
As a user of the scRNA Core Cell Extraction pipeline, I would like to have bed verification support in Limber for the "LRC PBMC Bank" plate to the "LRC Bank Seq" / "LRC Bank Spare" transfer, to try to avoid lab mistakes.

**Who are the primary contacts for this story**
Lesley, Katy

**Who is the nominated tester for UAT**
Lesley

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Robot name: Hamilton Cell Banking [not doing robot barcode verification at the moment]
- [ ] Robot type: Hamilton STAR [not doing robot barcode verification at the moment]
- [x] Source plate scanned into bed 12
- [x] For source plate, verify that it is of purpose "LRC PBMC Bank"
- [x] Destination tube rack barcodes scanned into beds 15 and 14
- [x] For destination racks, verify that they are in the correct beds by looking at the purposes of the tubes they contain:
    - [x] Purpose "LRC Bank Seq" should be in bed 15
    - [x] Purpose "LRC Bank Spare" should be in bed 14
- [x] Validate that destination tube rack barcodes are the same as those uploaded in the tube rack scan files (see https://github.com/sanger/limber/issues/1314)

**Dependencies**
This story is blocked by the following dependencies:
- https://github.com/sanger/limber/issues/1314

**Comment by Lesley84** [comment]
Can we have multiple destination plates (sequencing and contingency)?

**Comment by KatyTaylor** [comment]
Update on last comment: 
Robot software will prompt user with which barcode goes in which bed. See https://github.com/sanger/limber/issues/1362#issuecomment-1680501001 for summary

**Comment by KatyTaylor** [comment]
N.B. the robot to be used for this doesn't yet have barcodes stuck on it, so that will have to be done before bed verification can happen.

**Comment by KatyTaylor** [comment]
Updated to remove reference to multiple source plates, as there will now only be one.

**Comment by Lesley84** [comment]
Happy with this story. We can print the bed labels, but would you be able to provide us with a robot label?

**Comment by andrewsparkes** [comment]
The tubes now have 2 values stored in their custom_metadata:
tube_rack_barcode -> the barcode of the rack the tube was in at the point of transfer from the PBMC Bank plate
tube_rack_position -> the position of the tube within the rack at the point of transfer from the PBMC Bank plate

So you can fetch the rack barcode from the tube custom_metadata

**Comment by yoldas** [comment]
[DPL-829 LRC PBMBC Bank to LRC Bank Seq/Spare Bed Verification](https://lucid.app/lucidchart/e47fde78-7eef-46fe-9710-f41c00a9e741/edit?viewport_loc=-154%2C-598%2C2183%2C1222%2C0_0&invitationId=inv_9121c264-9769-4b87-9a1b-3675d983c2f1) Lucid document.

**Comment by yoldas** [comment]
About the first two acceptance criteria. We are not doing the verification of the robot barcode at the moment as it would require having `created_with_robot => ROBOT_BARCODE` in the metadata of parent PBMC Bank plate, which was not written. Therefore, we do not have `require_robot` option in the robot configuration for `'Hamilton LRC PBMC Bank => LRC Bank Seq and LRC Bank Spare'.`

**Comment by yoldas** [comment]
There is a [merge request](https://gitlab.internal.sanger.ac.uk/psd/integration-suite/-/merge_requests/101) for the integration suite related to this story on GitLab. It needs to be merged into its parent branch there after we merge the PR of this story to develop.
